src/libxtp/gwbse/gwbse.cc

```diff
@@ -721,5 +724,60 @@ bool GWBSE::Evaluate() {
       << TimeStamp() << " GWBSE calculation finished " << flush;
   return true;
 }
+
+Eigen::MatrixXd GWBSE::PrepareHQP_forBSE(const Eigen::MatrixXd& Hqp) {
+
+  Index bse_vmax = _bseopt.homo;
+  Index bse_cmin = _bseopt.homo + 1;
+  Index bse_vtotal = bse_vmax - _bseopt.vmin + 1;
+  Index bse_ctotal = _bseopt.cmax - bse_cmin + 1;
+  Index hqp_size = bse_vtotal + bse_ctotal;
+  Index gwsize = _gwopt.qpmax - _gwopt.qpmin + 1;
+  Index RPAoffset = _bseopt.vmin - _gwopt.rpamin;
+  Eigen::MatrixXd Hqp_BSE = Eigen::MatrixXd::Zero(hqp_size, hqp_size);
+
+  // if bse range is smaller than qprange, cut out  Hqp block
+  if (_bseopt.vmin >= _gwopt.qpmin) {
+    Index start = _bseopt.vmin - _gwopt.qpmin;
+    if (_bseopt.cmax <= _gwopt.qpmax) {
+      Hqp_BSE = Hqp.block(start, start, hqp_size, hqp_size);
+    } else {
+      Hqp_BSE.block(0, 0, gwsize - start, gwsize - start) =
```

Comment:
instead of `block` use `topLeftCorner` when you start at 0,0

Reply:
Done